firebase_admin/app_check.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Firebase App Check module."""
+
+# ASK(lahiru) Do I need to add these imports to the requirements file?
+from typing import Any, Dict, List
+import jwt
+from jwt import PyJWKClient, DecodeError
+from firebase_admin import _utils
+
+_APP_CHECK_ATTRIBUTE = '_app_check'
+
+def _get_app_check_service(app) -> Any:
+    return _utils.get_app_service(app, _APP_CHECK_ATTRIBUTE, _AppCheckService)
+
+# should i accept an app (design doc doesn't have one) or just always make it none
+def verify_token(token: str, app=None) -> Dict[str, Any]:
+    """Verifies a Firebase App Check token.
+
+    Args:
+        token: A token from App Check.
+        app: An App instance (optional).
+
+    Returns:
+        Dict[str, Any]: A token's decoded claims
+        if the App Check token is valid; otherwise, a rejected promise..
+    """
+    return _get_app_check_service(app).verify_token(token)
+
+class _AppCheckService:
+    """Service class that implements Firebase App Check functionality."""
+
+    _APP_CHECK_GCP_API_URL = 'https://firebaseappcheck.googleapis.com'
+    _APP_CHECK_BETA_JWKS_RESOURCE = '/v1beta/jwks'
+
+    def __init__(self, app):
```

Comment:
Unsure what I should include in this constructor, or even if I should include one